app/models/manageiq/providers/lenovo/physical_infra_manager/refresh_parser.rb

```diff
@@ -69,23 +88,43 @@ def parse_firmware(firmware, uuid)
       }
     end
 
+    def parse_hardware(node)
+      return nil if node.memoryModules.blank? && node.processors.blank?
+
+      new_result = {
+        :memory_mb       => 0,
+        :cpu_total_cores => 0,
+        :cpu_speed       => 0
+      }
+
+      node.memoryModules.each do |mem|
+        new_result[:memory_mb] += mem['capacity'] * 1024
+      end
+
+      node.processors.each do |pr|
+        new_result[:cpu_total_cores] += pr['cores']
+        new_result[:cpu_speed] = pr['speed'].to_i if pr['speed'].to_i > new_result[:cpu_speed]
+      end
+      new_result
+    end
+
     def parse_nodes(node)
       new_result = {
-        :type          => ManageIQ::Providers::Lenovo::PhysicalInfraManager::PhysicalServer.name,
-        :name          => node.name,
-        :ems_ref       => node.uuid,
-        :uid_ems       => @ems.uid_ems,
-        :hostname      => node.hostname,
-        :product_name  => node.productName,
-        :manufacturer  => node.manufacturer,
-        :machine_type  => node.machineType,
-        :model         => node.model,
-        :serial_number => node.serialNumber,
-        :uuid          => node.uuid,
-        :FRU           => node.FRU,
-        :macAddresses  => node.macAddress.split(",").flatten,
-        :ipv4Addresses => node.ipv4Addresses.split.flatten,
-        :ipv6Addresses => node.ipv6Addresses.split.flatten
+        :type                  => ManageIQ::Providers::Lenovo::PhysicalInfraManager::PhysicalServer.name,
```

Comment:
This PR should be closed and splited in two others:

1. Parse hardware informations(https://github.com/ManageIQ/manageiq-providers-lenovo/pull/39)
2. Add missing attributes